test/integration/arbitrum-mainnet/Loans.fork.t.sol

```diff
@@ -463,21 +482,160 @@ contract LoansForkTest is LoansTestBase {
+        skip(pair.durations[1] / 2);
+        // Move price above call strike using lib
+        PriceManipulationLib.movePriceUpPastCallStrike(
+            vm,
+            address(pair.swapperUniV3.uniV3SwapRouter()),
+            whale,
+            pair.cashAsset,
+            pair.underlying,
+            pair.oracle,
+            position.callStrikePercent,
+            POOL_FEE_TIER,
+            AMOUNT_FOR_CALL_STRIKE
+        );
+
+        skip(pair.durations[1] / 2);
+
+        (uint expectedTakerWithdrawal,) =
+            pair.takerNFT.previewSettlement(position, pair.oracle.currentPrice());
+
+        // Total cash = taker withdrawal + loan repayment
+        // Convert expected total cash to underlying at current price
+        uint expectedUnderlyingOut =
+            pair.oracle.convertToBaseAmount(expectedTakerWithdrawal + loanAmount, pair.oracle.currentPrice());
+
+        uint userUnderlyingBefore = pair.underlying.balanceOf(user);
+
+        // Close loan and settle
+        closeAndCheckLoan(loanId, loanAmount, loanAmount + expectedTakerWithdrawal, 0);
+
+        // Check provider's withdrawable amount
+        uint providerWithdrawable = position.providerNFT.getPosition(position.providerId).withdrawable;
+        assertEq(providerWithdrawable, 0); // everything to user
+
+        // Check user's underlying balance change against calculated expected amount
+        assertApproxEqAbs(
+            pair.underlying.balanceOf(user) - userUnderlyingBefore,
+            expectedUnderlyingOut,
+            expectedUnderlyingOut * slippage / 100 // within slippage% of expected
+        );
+    }
+
+    function testSettlementPriceBelowPutStrike() public {
+        // Create provider offer & open loan
+        uint offerId = createProviderOffer(pair, callstrikeToUse, offerAmount, pair.durations[1]);
+        (uint loanId,, uint loanAmount) = openLoan(pair, user, underlyingAmount, 0.3e6, offerId);
+
+        ICollarTakerNFT.TakerPosition memory position = pair.takerNFT.getPosition(loanId);
+
+        skip(pair.durations[1] / 2);
+
+        // Move price below put strike
+        PriceManipulationLib.movePriceDownPastPutStrike(
+            vm,
+            address(pair.swapperUniV3.uniV3SwapRouter()),
+            whale,
+            pair.cashAsset,
+            pair.underlying,
+            pair.oracle,
+            position.putStrikePercent,
+            POOL_FEE_TIER,
+            AMOUNT_FOR_PUT_STRIKE
+        );
+
+        skip(pair.durations[1] / 2);
+        uint userUnderlyingBefore = pair.underlying.balanceOf(user);
+        // Close loan
+        closeAndCheckLoan(loanId, loanAmount, loanAmount, 0);
+        uint expectedUnderlying = pair.oracle.convertToBaseAmount(loanAmount, pair.oracle.currentPrice());
+        assertApproxEqAbs(
+            pair.underlying.balanceOf(user) - userUnderlyingBefore,
+            expectedUnderlying,
+            expectedUnderlying * slippage / 100
+        );
```

Comment:
the common part can probably be refactored (up to the last bit with specific values)

Reply:
I was thinking about it , but the specific value changes of each as well as the changes in assertion make it kinda difficult to refactor without just making it unecessary, if you see it more clearly lmk